### ISSUE TITLE
CPU Usage

### ISSUE THREAD

**Issue by jsalbin** [Rappsilber-Laboratory/XiSearch#55]
Thank you for a great tool. I have been having an issue with XiSearch v1.7.6.1 in which CPU usage frequently spikes to 100%, freezing XiSearch and overall making it impractical to run to completion, as the frequency of these spikes will eventually increase to a point at which XiSearch is mostly frozen. I have 12 threads and 64GB RAM on a Lenovo P73 running Windows 10 Pro. No combination of thread and RAM allocation seems to affect the CPU usage pattern. I do not know Java, but in searching, it seems like it is probably a Garbage Collection issue. Is there a simple fix for this?

**Comment by lutzfischer** [comment]
Yes, could be problem with garbage collection. You could have a look at the feedback tab; there is some info on used memory.
But xiSEARCH should cause pretty high CPU-usage by default - as it is configured to start n-1 search threads - where n is the number of available logical cores.

If you have not tried yet you can change the `-Xmx` paramater in the start-file (e.g. `startXiWindows.bat`) to something substantially bigger.
Another thing you could try is to reduce the number of threads. That would reduce the total amount of memory used during search. 
Also you can play wit a customsetting: maxpeakcandidates - you can set that to something like 10000 ( `maxpeakcandidates:10000` ). This might reduce the sensitivity a bit but should also limit the memory usage.
How many Proteins are you searching?
